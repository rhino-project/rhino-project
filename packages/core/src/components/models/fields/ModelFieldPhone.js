import PropTypes from 'prop-types';
import FieldPhone from 'rhino/components/forms/fields/FieldPhone';
import { useGlobalComponent } from 'rhino/hooks/overrides';

const ModelFieldPhoneBase = ({ model, ...props }) => {
  return <FieldPhone {...props} />;
};

ModelFieldPhoneBase.propTypes = {
  model: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  path: PropTypes.string.isRequired
};

const ModelFieldPhone = (props) =>
  useGlobalComponent('ModelFieldPhone', ModelFieldPhoneBase, props);

export default ModelFieldPhone;
