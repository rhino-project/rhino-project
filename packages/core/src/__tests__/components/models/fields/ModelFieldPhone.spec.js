import { sharedFieldTests } from './sharedFieldTests';
import ModelFieldPhone from 'rhino/components/models/fields/ModelFieldPhone';

describe('ModelFieldPhone', () => {
  sharedFieldTests(ModelFieldPhone);
});
