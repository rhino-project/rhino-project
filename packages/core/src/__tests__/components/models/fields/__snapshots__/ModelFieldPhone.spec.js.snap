// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModelFieldPhone > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelFieldPhone > should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelFieldPhone > should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
